### PR TITLE
Extract Go snippets

### PR DESCRIPTION
**Pull request by erdemyerebasmaz**
This PR adds Go snippets to
- #72 

##### TODO:
- [ ] Add CI for Go snippets

### CODE REVIEW

File: snippets/go/go.sum [hunk 1]
```diff
@@ -0,0 +1,2 @@
+github.com/breez/breez-sdk-go v0.2.7 h1:9Kemh1y/T2y6y9eIR3ZYYbfYNQYyJ+UfyIoI8xmar90=
```
Comment (JssDWt) [inline snippets/go/go.sum:1]: let's gitignore this file to make it easier to switch versions

File: snippets/go/main.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package main
```
Comment (JssDWt) [inline snippets/go/main.go:1]: There is a binary file called `main` committed

File: snippets/go/send_payment.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+	"github.com/breez/breez-sdk-go/breez_sdk"
+)
+
+func SendPayment() {
+	// ANCHOR: send-payment
+	bolt11 := "bolt11 invoice"
+	// The `amountMsat` param is optional and should only passed if the bolt11 doesn't specify an amount.
+	// The amountMsat is required in case an amount is not specified in the bolt11 invoice'.
+	amountMsat := uint64(3000000)
```
Comment (JssDWt) [inline snippets/go/send_payment.go:14]:
```suggestion
	amountMsat := uint64(3_000_000)
```

File: snippets/go/send_onchain.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	// ANCHOR: get-current-reverse-swap-min-max
+	log.Printf("Minimum amount, in sats: %v", currentFees.Min)
+	log.Printf("Maximum amount, in sats: %v", currentFees.Max)
+	// ANCHOR_END: get-current-reverse-swap-min-max
+}
+
+func StartReverseSwap() {
+	// ANCHOR: start-reverse-swap
+	destinationAddress := "bc1.."
+	sendAmountSat := uint64(50000)
```
Comment (JssDWt) [inline snippets/go/send_onchain.go:31]:
```suggestion
	sendAmountSat := uint64(50_000)
```

File: snippets/go/send_onchain.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+import (
+	"log"
+
+	"github.com/breez/breez-sdk-go/breez_sdk"
+)
+
+func GetCurrentFees() {
+	// ANCHOR: estimate-current-reverse-swap-total-fees
+	sendAmountSat := uint64(50000)
```
Comment (JssDWt) [inline snippets/go/send_onchain.go:11]:
```suggestion
	sendAmountSat := uint64(50_000)
```

File: snippets/go/receive_onchain.go [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+		if result, err := sdk.Refund(refundRequest); err == nil {
+			log.Printf("%v", result)
+		}
+	}
+	// ANCHOR_END: execute-refund
+}
+
+func GetChannelOpeningFees() {
+	// ANCHOR: get-channel-opening-fees
+	amountMsat := uint64(10000)
```
Comment (JssDWt) [inline snippets/go/receive_onchain.go:54]: this can be passed in as a parameter. Or make it 3_000_000.

**Review by JssDWt — COMMENTED** [review]
left a few comments.

**Review by dangeross — CHANGES_REQUESTED** [review]
Looks good, just a small anchor fix needed

**Comment by erdemyerebasmaz** [comment]
Addressed feedback.
- Removed & added binary & dependency checksum files to .gitignore
- Added underscore as a separator in number literals
- Corrected get-channel-opening-fees anchor_end
- Farewell to "Hello World"

**Review by JssDWt — APPROVED** [review]
LGTM!

**Review by dangeross — APPROVED** [review]
🚀

**Pull request merged**